qiskit/dagcircuit/collect_blocks.py

```diff
@@ -170,13 +212,28 @@ def not_filter_fn(node):
             if matching_block:
                 matching_blocks.append(matching_block)
 
+        # If the option split_layers is set, refine blocks by splitting them into layers
+        # of non-overlapping instructions (in other words, into depth-1 sub-blocks).
+        if split_layers:
+            tmp_blocks = []
+            for block in matching_blocks:
+                tmp_blocks.extend(split_block_into_layers(block))
+            matching_blocks = tmp_blocks
+
         # If the option split_blocks is set, refine blocks by splitting them into sub-blocks over
         # disconnected qubit subsets.
         if split_blocks:
-            split_blocks = []
+            tmp_blocks = []
             for block in matching_blocks:
-                split_blocks.extend(BlockSplitter().run(block))
-            matching_blocks = split_blocks
+                tmp_blocks.extend(BlockSplitter().run(block))
+            matching_blocks = tmp_blocks
+
+        # If we are collecting from the back, both the order of the blocks
+        # and the order of nodes in each block should be reversed.
+        if self._collect_from_back:
+            matching_blocks = matching_blocks[::-1]
+            for i, block in enumerate(matching_blocks):
+                matching_blocks[i] = block[::-1]
```

Comment:
I kept coming back to this block and thinking maybe it be easier to replace this with:

```suggestion
            matching_blocks = [block[::-1] for block in matching_blocks[::-1]]
```